api/include/opentelemetry/trace/propagation/http_trace_context.h

```diff
@@ -0,0 +1,325 @@
+                          nostd::string_view trace_description);
+
+  void Inject(Setter setter, T &carrier, const context::Context &context) noexcept override
+  {
+    SpanContext span_context = GetCurrentSpan(context)->GetContext();
+    if (!span_context.IsValid())
+    {
+      return;
+    }
+    InjectImpl(setter, carrier, span_context);
+  }
+
+  context::Context Extract(Getter getter,
+                           const T &carrier,
+                           context::Context &context) noexcept override
+  {
+    SpanContext span_context    = ExtractImpl(getter, carrier);
+    nostd::string_view span_key = "current-span";
+    nostd::shared_ptr<Span> sp{new DefaultSpan(span_context)};
+    return context.SetValue(span_key, sp);
+  }
+
+  static Span *GetCurrentSpan(const context::Context &context)
+  {
+    const nostd::string_view span_key = "current-span";
+    context::Context ctx(context);
+    nostd::shared_ptr<Span> span = nostd::get<nostd::shared_ptr<Span>>(ctx.GetValue(span_key));
+    return (span.get());
+  }
+
+  static TraceId GenerateTraceIdFromString(nostd::string_view trace_id)
+  {
+    const char *trc_id = trace_id.begin();
+    uint8_t buf[kTraceIdBytes / 2];
+    int tmp;
+    for (int i = 0; i < kTraceIdBytes; i++)
+    {
+      tmp = CharToInt(*trc_id);
+      if (tmp < 0)
+      {
+        for (int j = 0; j < kTraceIdBytes / 2; j++)
+        {
+          buf[j] = 0;
+        }
+        return TraceId(buf);
+      }
+      if (i % 2 == 0)
+      {
+        buf[i / 2] = tmp * 16;
+      }
+      else
+      {
+        buf[i / 2] += tmp;
+      }
+      trc_id++;
+    }
+    return TraceId(buf);
+  }
+
+  static SpanId GenerateSpanIdFromString(nostd::string_view span_id)
+  {
+    const char *spn_id = span_id.begin();
+    uint8_t buf[kSpanIdBytes / 2];
+    int tmp;
+    for (int i = 0; i < kSpanIdBytes; i++)
+    {
+      tmp = CharToInt(*spn_id);
+      if (tmp < 0)
+      {
+        for (int j = 0; j < kSpanIdBytes / 2; j++)
+        {
+          buf[j] = 0;
+        }
+        return SpanId(buf);
+      }
+      if (i % 2 == 0)
+      {
+        buf[i / 2] = tmp * 16;
+      }
+      else
+      {
+        buf[i / 2] += tmp;
+      }
+      spn_id++;
+    }
+    return SpanId(buf);
+  }
+
+  static TraceFlags GenerateTraceFlagsFromString(nostd::string_view trace_flags)
+  {
+    uint8_t buf;
+    int tmp1 = CharToInt(trace_flags[0]);
+    int tmp2 = CharToInt(trace_flags[1]);
+    if (tmp1 < 0 || tmp2 < 0)
+      return TraceFlags(0);  // check for invalid char
+    buf = tmp1 * 16 + tmp2;
+    return TraceFlags(buf);
+  }
+
+private:
+  static uint8_t CharToInt(char c)
+  {
+    if (c >= '0' && c <= '9')
+    {
+      return (int)(c - '0');
+    }
+    else if (c >= 'a' && c <= 'f')
+    {
+      return (int)(c - 'a' + 10);
+    }
+    else if (c >= 'A' && c <= 'F')
+    {
+      return (int)(c - 'A' + 10);
+    }
+    else
+    {
+      return -1;
+    }
+  }
+
+  // TODO:
```

Comment:
In my opinion, merging these TraceState TODO comments is unnecessary, since you already have a branch that implements all of the changes that will be merged shortly after this PR. I think the code would be cleaner without these intermediate comments. Some might disagree with me though, so it's up to you.

Reply:
I have cleaned up the trace state related TODOs in Http Trace Context and Span context